ci-conda.Dockerfile

```diff
@@ -1,21 +1,95 @@
 ARG CUDA_VER=notset
 ARG LINUX_VER=notset
+ARG BASE_IMAGE=notset
 ARG PYTHON_VER=notset
 ARG YQ_VER=notset
 ARG AWS_CLI_VER=notset
 ARG MINIFORGE_VER=notset
+ARG CPU_ARCH=notset
+ARG NVARCH=notset
 
 FROM condaforge/miniforge3:${MINIFORGE_VER} AS miniforge-upstream
-FROM nvidia/cuda:${CUDA_VER}-base-${LINUX_VER} AS miniforge-cuda
+FROM ${BASE_IMAGE} AS miniforge-cuda
 
 ARG LINUX_VER
 ARG PYTHON_VER
+ARG CUDA_VER
+ARG CPU_ARCH
+ARG NVARCH
 ARG DEBIAN_FRONTEND=noninteractive
 ENV PATH=/opt/conda/bin:$PATH
 ENV PYTHON_VERSION=${PYTHON_VER}
 
 SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
 
+ENV NVARCH=${NVARCH}
+ENV CUDA_VERSION=${CUDA_VER}
+
+# Install CUDA based on the nvidia/cuda templates
+RUN <<EOF
+case "${LINUX_VER}" in
+  "ubuntu"*)
+    # Extract Ubuntu version number using bash variable expansion
+    UBUNTU_VERSION=${LINUX_VER#ubuntu}
+    UBUNTU_VERSION=${UBUNTU_VERSION//./}
+    PKG_CUDA_VER="$(echo ${CUDA_VER} | cut -d '.' -f1,2 | tr '.' '-')"
```

Comment:
Minor, but this is declared twice with the same logic, can we declare it once in the RUN block?